spec/decorators/usps_decorator_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+class DummyController < ApplicationController
+  def initialize
+    @session = {}
+  end
+
+  def set(session)
+    @session = session
+  end
+
+  def user_session
+    @session
+  end
+end
+
+RSpec.describe UspsDecorator do
+  subject(:decorator) do
+    user = create(
+      :user,
+      :signed_up,
+      profiles: [build(:profile, :active, :verified, pii: { first_name: 'Jane' })]
+    )
+    controller = DummyController.new
+    idv_session = Idv::Session.new(controller.user_session, user)
+    UspsDecorator.new(idv_session: idv_session)
+  end
+
+  describe '#title' do
+    context 'a letter has not been sent' do
+      it 'provides text that indicates that' do
+        allow(subject).to receive(:letter_already_sent?).and_return(false)
+        expect(subject.title).to eq(
+          I18n.t('idv.titles.mail.verify')
+        )
+      end
+    end
+
+    context 'a letter has been sent' do
+      it 'provides text that indicates that' do
+        allow(subject).to receive(:letter_already_sent?).and_return(true)
+        expect(subject.title).to eq(
+          I18n.t('idv.titles.mail.resend')
+        )
+      end
+    end
+  end
+
+  describe '#button' do
+    context 'a letter has not been sent' do
+      it 'provides text that indicates that' do
+        allow(subject).to receive(:letter_already_sent?).and_return(false)
+        expect(subject.button).to eq(
+          I18n.t('idv.buttons.mail.send')
+        )
+      end
+    end
+
+    context 'a letter has been sent' do
+      it 'provides text that indicates that' do
+        allow(subject).to receive(:letter_already_sent?).and_return(true)
```

Comment:
I think it would be easier to just not stub this in each lets and use `let`s to set the IDV verification mechanism in each test